pkg/cmd/pulumi/pulumi.go

```diff
@@ -186,55 +267,94 @@ func NewPulumiCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVar(
 		&color, "color", "auto", "Colorize output. Choices are: always, never, raw, auto")
 
-	// Common commands:
-	//     - Getting Started Commands:
-	cmd.AddCommand(newNewCmd())
-	//     - Deploy Commands:
-	cmd.AddCommand(newUpCmd())
-	cmd.AddCommand(newPreviewCmd())
-	cmd.AddCommand(newDestroyCmd())
-	cmd.AddCommand(newWatchCmd())
-	//     - Stack Management Commands:
-	cmd.AddCommand(newStackCmd())
-	cmd.AddCommand(newConfigCmd())
-	//     - Service Commands:
-	cmd.AddCommand(newLoginCmd())
-	cmd.AddCommand(newLogoutCmd())
-	cmd.AddCommand(newWhoAmICmd())
-	//     - Policy Management Commands:
-	cmd.AddCommand(newPolicyCmd())
-	//     - Advanced Commands:
-	cmd.AddCommand(newCancelCmd())
-	cmd.AddCommand(newImportCmd())
-	cmd.AddCommand(newRefreshCmd())
-	cmd.AddCommand(newStateCmd())
-	//     - Other Commands:
-	cmd.AddCommand(newLogsCmd())
-	cmd.AddCommand(newPluginCmd())
-	cmd.AddCommand(newVersionCmd())
-	cmd.AddCommand(newConsoleCmd())
-	cmd.AddCommand(newAboutCmd())
-	cmd.AddCommand(newSchemaCmd())
-	cmd.AddCommand(newOrgCmd())
-
-	// Less common, and thus hidden, commands:
-	cmd.AddCommand(newGenCompletionCmd(cmd))
-	cmd.AddCommand(newGenMarkdownCmd(cmd))
-
-	// We have a set of commands that are still experimental and that are hidden unless PULUMI_EXPERIMENTAL is set
-	// to true.
-	cmd.AddCommand(newQueryCmd())
-	cmd.AddCommand(newConvertCmd())
-
-	// We have a set of options that are useful for developers of pulumi that are hidden unless PULUMI_DEBUG_COMMANDS is
-	// set to true.
-	cmd.PersistentFlags().StringVar(&tracingHeaderFlag, "tracing-header", "",
-		"Include the tracing header with the given contents.")
+	setCommandGroups(cmd, []commandGroup{
+
+		// Common commands:
+		{
+			Name: "Deployment Commands",
+			Commands: []*cobra.Command{
+				newNewCmd(),
+				newConfigCmd(),
+				newUpCmd(),
+				newDestroyCmd(),
+				newPreviewCmd(),
+				newCancelCmd(),
+			},
+		},
+		{
+			Name: "Stack Management Commands",
+			Commands: []*cobra.Command{
+				newStackCmd(),
+				newConsoleCmd(),
+				newImportCmd(),
+				newRefreshCmd(),
+				newStateCmd(),
+			},
+		},
+		{
+			Name: "Service Commands",
+			Commands: []*cobra.Command{
+				newLoginCmd(),
+				newLogoutCmd(),
+				newWhoAmICmd(),
+				newOrgCmd(),
+			},
+		},
+		{
+			Name: "Policy Management Commands",
```

Comment:
```suggestion
			Name: "Policy Commands",
```
sounds enough?

Reply:
I think it makes sense to keep `Policy Management Commands` because without it, a user that has never used policy may not get the impression that `Policy Commands` is a complex feature and may skip over it.

We could change it to `Policy as Code Commands` (given that's the full name of the product) to make it more intriguing.

Reply:
How do you know that "Policy Commands" is perceived as more complex than "Policy Management Commands"?

If a user has never heard of Policy as Code in Pulumi, I suspect this command is not the best place to start anyway.

Reply:
Unfortunately, our current policy help command has no Long description

```
Manage resource policies

Usage:
```

We can leave it as Policy Commands since the help menu is currently lacking, but I'd recommend we change it once the help text is up to standard.

Policy is really boring. We need a clickbait heading like `Policy as Code Commands` to get the user excited and interested in viewing the help menu for `pulumi policy` rather than reusing a bland term like Policy that every infrastructure product has beat the meaning out of.

The problem is that we can't control where users start. We can only make it less painful. I tried starting using `pulumi import` via `pulumi import --help` and the help text was so useless that I just didn't continue to use pulumi for my use-case because I had already hit so many pain points that I had lost interest.

If they view the help menu, we can provide them a link to the docs and we can have more control over the experience to try and lead to a positive outcome.